apps/api/src/app/workflows-v2/usecases/get-workflow/get-workflow.usecase.ts

```diff
@@ -0,0 +1,97 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import defaults from 'json-schema-defaults';
+
+import { EnvironmentRepository, NotificationTemplateRepository } from '@novu/dal';
+import { Schema } from '@novu/framework';
+import { ControlVariablesLevelEnum } from '@novu/shared';
+
+import { ControlValuesEntity, ControlValuesRepository, NotificationStepEntity } from '@novu/dal/src';
+import { GetWorkflowCommand } from './get-workflow.command';
+import { WorkflowResponseDto } from '../../dto/workflow.dto';
+import { WorkflowTemplateGetMapper } from '../../mappers/workflow-template-get-mapper';
+
+@Injectable()
+export class GetWorkflowUseCase {
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private environmentRepository: EnvironmentRepository,
+    private controlValuesRepository: ControlValuesRepository
+  ) {}
+  async execute(command: GetWorkflowCommand): Promise<WorkflowResponseDto> {
+    await this.validateEnvironment(command);
+    const notificationTemplateEntity = await this.notificationTemplateRepository.findByIdQuery({
+      id: command._workflowId,
+      environmentId: command.user.environmentId,
+    });
+
+    if (notificationTemplateEntity === null || notificationTemplateEntity === undefined) {
+      throw new BadRequestException(`Workflow not found with id: ${command._workflowId}`);
+    }
+    const stepIdToControlValuesMap = this.getControlsValuesMap(notificationTemplateEntity.steps, command);
+    const workflow = WorkflowTemplateGetMapper.toResponseWorkflowDto(notificationTemplateEntity);
+    //
+    return { ...workflow, steps: stepDtos };
+  }
+
+  private isContainValues(
+    stepControlsVariables: ControlValuesEntity | null
+  ): stepControlsVariables is ControlValuesEntity & { controls: Record<string, unknown> } {
+    if (!stepControlsVariables?.controls) {
+      return false;
+    }
+
+    return Object.keys(stepControlsVariables?.controls).length > 0;
+  }
+
+  private async validateEnvironment(command: GetWorkflowCommand) {
+    const environment = await this.environmentRepository.findOne({ _id: command.user.environmentId });
+
+    if (!environment) {
+      throw new BadRequestException('Environment not found');
+    }
+  }
+
+  private async getControlsValuesMap(
+    steps: NotificationStepEntity[],
+    command: GetWorkflowCommand
+  ): Promise<{ [key: string]: ControlValuesEntity }> {
+    const acc: { [key: string]: ControlValuesEntity } = {};
+
+    for (const step of steps) {
+      let controlValuesEntity = await this.buildControlVariablesforStep(step, command);
+      if (controlValuesEntity) {
+        acc[step._templateId] = controlValuesEntity;
+      }
+    }
+
+    return acc; // Return the accumulated results
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** Please remove the redundant comment.‏

Reply:
need to make my chat gpt stop commenting :)